pkg/devices.go

```diff
@@ -0,0 +1,210 @@
+package pkg
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	cdispec "github.com/container-orchestrated-devices/container-device-interface/specs-go"
+	spec "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	root = "/etc/cdi"
+)
+
+var (
+	defaultLogger = logrus.New()
+)
+
+func init() {
+	defaultLogger.SetLevel(logrus.WarnLevel)
+}
+
+func collectCDISpecs(log *logrus.Logger) (map[string]*cdispec.Spec, error) {
+	var files []string
+	vendor := make(map[string]*cdispec.Spec)
+
+	err := filepath.Walk(root, func(path string, info os.FileInfo, err error) error {
+		if info == nil || info.IsDir() {
+			log.Infof("CDI: Skipped directory %q", path)
+			return nil
+		}
+
+		if filepath.Ext(path) != ".json" {
+			log.Infof("CDI: Skipped non json terminated file %q", path)
```

Comment:
Looks to me this is just printing, the append is still happening below

Reply:
Good catch!